contracts/test/RefinanceTest.t.sol

```diff
@@ -306,4 +306,170 @@ contract RefinanceTest is AddressRegistry, TestUtils {
         assertEq(details[6], 0);
     }
 
+    function test_refinance_samePrincipal() external {
+        bytes[] memory calls = new bytes[](1);
+        calls[0] = abi.encodeWithSelector(Refinancer.setPaymentsRemaining.selector, 6);
+
+        borrower.loan_proposeNewTerms(address(loanV3), address(refinancer), deadline, calls);
+        DebtLocker(loanV3.lender()).acceptNewTerms(address(refinancer), deadline, calls, 0);
+
+        assertEq(loanV3.principal(),           1_000_000_000000);
+        assertEq(loanV3.endingPrincipal(),     1_000_000_000000);
+        assertEq(loanV3.interestRate(),        0.12e18);
+        assertEq(loanV3.paymentsRemaining(),   6);
+        assertEq(loanV3.paymentInterval(),     30 days);
+        assertEq(loanV3.delegateFee(),         205_479452);  // 1,000,000 * 0.5% * 30 / 365
+        assertEq(loanV3.treasuryFee(),         205_479452);
+        assertEq(loanV3.refinanceCommitment(), 0);
+        assertEq(loanV3.drawableFunds(),       0);
+        assertEq(loanV3.claimableFunds(),      0);
+
+        vm.warp(start + 60 days);
+
+        ( uint256 principalPortion, uint256 interestPortion, uint256 delegateFee, uint256 treasuryFee ) = loanV3.getNextPaymentBreakdown();
+
+        assertEq(principalPortion, 0);
+        assertEq(interestPortion,  9_863_013698); // 1,000,000 * 12% * 30 / 365
+        assertEq(delegateFee,      205_479452);
+        assertEq(treasuryFee,      205_479452);
+
+        uint256 totalPayment = principalPortion + interestPortion + delegateFee + treasuryFee;
+
+        erc20_mint(USDC, 9, address(borrower), totalPayment);
+
+        borrower.erc20_approve(USDC, address(loanV3), totalPayment);
+        borrower.loan_makePayment(address(loanV3), totalPayment);
+
+        assertEq(loanV3.drawableFunds(),      0);
+        assertEq(loanV3.claimableFunds(),     interestPortion);
+        assertEq(loanV3.nextPaymentDueDate(), start + 90 days);
+        assertEq(loanV3.principal(),          1_000_000_000000);
+        assertEq(loanV3.paymentsRemaining(),  5);
+        
```

Comment:
Add erc20 balance assertions for loan and liquidity locker before and after claim

Reply:
Goes for all tests